rsmtool/fairness_utils.py

```diff
@@ -0,0 +1,287 @@
+    # This makes sure that the order is consistent and reproducible across runs
+    # when there are more than
+    # one group with the maximum number of occurrences. 
+    df_groups_by_size = pd.DataFrame(group_values.value_counts()).reset_index()
+    df_groups_by_size.columns = ['group_name', 'count']
+    df_groups_by_size_sorted = df_groups_by_size.sort_values(['count', 'group_name'],
+                                                            ascending=[False, True])
+    groups_by_size = df_groups_by_size_sorted['group_name'].values
+
+    #convert to category and reorder
+    group_category = group_values.astype("category")
+    group_ordered_category = group_category.cat.reorder_categories(groups_by_size,
+                                                                   ordered=True)
+    return group_ordered_category
+    
+
+
+
+def get_coefficients(fit, base_category):
+    
+    """
+    Extract estimates, significance, and confidence intervals
+    for members of the group of interest. The names of the 
+    predictors are processed to remove the prefix added by Statmodels.
+    The name of the base category is added in parenthesis to the Intercept. 
+
+    Parameters:
+    -----------
+    fit: pandas.stats.ols.OLS
+        Linear regression object fitted using statsmodels
+
+    base_category: str
+        Name of the group used as reference category when fitting the model
+
+    Returns:
+    -------
+    df_results: pandas DataFrame
+        A dataframe with rows for each category and the following columns:
+        - estimate
+        - P>[t]
+        - Lower and upper boundaries for 95% confidence intervals
+    """
+        
+    # extract the data we need
+    df_results = pd.concat([fit.params, fit.pvalues, fit.conf_int()], axis=1)
+        
+    df_results.columns = ['estimate', 'P>[t]', '[0.025', '0.975]']
+        
+    # identify the rows we are interested in
+    groups = ['Intercept'] + [v for v in df_results.index if 'group' in v]
+        
+    df_results = df_results.loc[groups]
+        
+    # rename the rows
+    df_results.index = [v.split('.')[1].strip(']') 
+                            if not v == 'Intercept' 
+                            else 'Intercept ({})'.format(base_category)
+                            for v in df_results.index]
+        
+    return df_results
+
+
+
+
+def write_fairness_results(fit_dictionary,
+                           fairness_container,
+                           group,
+                           output_dir,
+                           experiment_id, 
+                           file_format):
+    """
+    Save the results of fairness analysis to disk.
+
+    Parameters
+    ----------
+    fit_dictionary: dict
+        A dictionary of fitted models generated by `get_fairness_analysis()`
+    fairness_container: container.DataContainer
+        A data container with the results of fairness analysis generated by `get_fairness_analysis()`
```

Comment:
```suggestion
        A data container with the results of fairness analysis generated by `get_fairness_analyses()`
```

Reply:
Oh, I fixed it in another branch. Let me fix it here too :)